src/store.rs

```diff
@@ -45,42 +49,71 @@ pub struct InnerStore {
     waker: Notify,
 }
 
-impl Deref for Store {
-    type Target = InnerStore;
-
-    fn deref(&self) -> &Self::Target {
-        &self.inner
-    }
+pub struct InnerStoreLocked<'a> {
+    state: MutexGuard<'a, State>,
+    waker: &'a Notify,
```

Comment:
store.lock() returns an InnerStoreLocked wrapper type, which acts like a transaction. While this struct is held, all operations occur on the locked store, ensuring no other changes can occur concurrently. This is particularly useful for methods like SET, where a value may need to be read and then updated—allowing you to avoid locking the store multiple times by reusing the locked state.

However, it’s important to release the InnerStoreLocked as soon as possible, since holding it for longer than necessary blocks access to the store’s state for other operations.